### PR TITLE
x86: add 6.1 testing kernel

### PR DESCRIPTION
**Pull request by coudu**
This PR adds support for testing kernel 6.1 for the x86 target.

Compile-tested: x86_64, x86_generic, x86_legacy, x86_geode
Run-tested: x86_64, x86_generic (qemu), x86_legacy (qemu)

### CODE REVIEW

**Comment by graysky2** [comment]
@coudu - Thank you for this work.  Did you compare the config-6.1 you are proposing here to the older version https://github.com/openwrt/openwrt/pull/12982 ?  for 64/config-6.1 anyway, your PR is missing quite a few symbols.

**Comment by coudu** [comment]
> @coudu - Thank you for this work. Did you compare the config-6.1 you are proposing here to the older version #12982 ? for 64/config-6.1

Thank you for pointing this out, I totally missed that there were a previous PR trying to add 6.1 testing to x86, I'll have a look a it to see if I can add progress that was made there.

> anyway, your PR is missing quite a few symbols.

Absolutely, do you want me to convert this PR to draft until those are added?

**Comment by graysky2** [comment]
I wouldn't do that.  Some feedback from @pkgadd or @PolynomialDivision?

**Comment by coudu** [comment]
Should be good for x86_64 now. I'm working on the 32bits subtargets, I'll be able to compile but to test on real hardware...

**Comment by hauke** [comment]
You can run the images in qemu using `./scripts/qemustart x86 generic`.

**Comment by coudu** [comment]
> You can run the images in qemu using `./scripts/qemustart x86 generic`.

Thanks for the tip, as I mainly build for the hardware I own, I usually test on real hardware and didn't notice there was a qemu starter script to test image.

For the geode subtarget to see if everything works as it should on real hardware, I've got an alix2c2 somewhere but can't find it at the moment :-[

**Comment by coudu** [comment]
Should also be good for x86_generic.

**Comment by graysky2** [comment]
Built and runs fine for me on x86/64/AMD Cezanne

**Comment by coudu** [comment]
Ok for all subtargets, now I'll build an image of each and test-run in qemu (only built the kernel as of now).
x86_64 is actually running on my main router (virtualized) with no problem.

**Comment by coudu** [comment]
OK for generic subtarget, geode will need some work in the modules video.mk

**Comment by coudu** [comment]
All subtagets compile and run-tested (qemu for generic and legacy) except geode.

Ready for review.

File: package/kernel/linux/modules/netdevices.mk [hunk 1]
```diff
@@ -1344,7 +1344,8 @@ define KernelPackage/mlx5-core
 	CONFIG_MLX5_MPFS=y \
 	CONFIG_MLX5_SW_STEERING=n \
 	CONFIG_MLX5_TC_CT=n \
-	CONFIG_MLX5_TLS=n
+	CONFIG_MLX5_TLS=n \
+	CONFIG_MLX5_VFIO_PCI=n@ge5.18
```
Comment (hauke) [inline package/kernel/linux/modules/netdevices.mk:1348]: You can always set this, if `CONFIG_MLX5_VFIO_PCI` does not exist on older kernel versions it does nothing.

Comment (coudu) [reply package/kernel/linux/modules/netdevices.mk]: Done. Thanks.

**Comment by coudu** [comment]
@graysky2 @hauke @PolynomialDivision 
as seen in the mentioned issue [#13361](https://github.com/openwrt/openwrt/issues/13361) newer kernel is needed for some devices.
Wouldn't it even be time to switch x86 target to 6.1 by default?
I can modify this PR to 6.1 by default if needed.

Thanks.

**Comment by trippleflux** [comment]
Run-Tested :  x86_64 Intel tigerlake
Kernel 6.1.53
Uptime：1d 18h 24m 24s

**Comment by oskarirauta** [comment]
config is missing:
 - Use nosuid,noexec mount options on devtmpfs (DEVTMPFS_SAFE) [N/y/?] (NEW)
 - DM audit events (DM_AUDIT) [N/y/?] (NEW)

**Pull request merged**